### PR TITLE
Add report feature, add random extensions

### LINKED ISSUE #15: Implement interview reports
## Summary
Core feature of the application should be possibility to create an interview report.

## Proposal
Implement feature to create an interview report based on rated questions which will contain average rating, average difficulty and interviewer note of each question.

### PR DESCRIPTION
**Pull request by rzerdik**
fixes #15 

added report creation feature, cleanup

### CODE REVIEW

**Comment by fpindej** [comment]
The functionality works on my end. I hope I tested everything 😄 

As for my take on this one: I wouldn't burn any more time and would ship it, since this is basically the entire functionality we need covered (afaik).

File: src/Client/Extensions/JSRuntimeExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿
+using Microsoft.JSInterop;
+
+namespace InterviewApp.Client.Extensions;
+
+public static class JSRuntimeExtensions
+{
+    public async static Task SaveAsAsync(this IJSRuntime js, string fileName, byte[] data)
+    {
+        await js.InvokeAsync<object>("saveAsFile", fileName, Convert.ToBase64String(data));
```
Comment (fpindej) [inline src/Client/Extensions/JSRuntimeExtensions.cs]: Expression body method

**Review by fpindej — APPROVED** [review]

File: src/Client/Dialogs/ReportDialog.razor [hunk 2]
```diff
@@ -0,0 +1,9 @@
+﻿<MudDialog Class="report-dialog">
+    <DialogContent>
+        <MudTextFieldString @bind-Value="ReportText" ReadOnly="true" Immediate="true" Lines="30" Label="Report" Class="report" Variant="Variant.Outlined" />
+    </DialogContent>
+    <DialogActions>
+        <MudButton Color="Color.Default" OnClick="Cancel">Close</MudButton>
+        <MudButton Color="Color.Primary" OnClick="Download">Download</MudButton>
+    </DialogActions>
+</MudDialog>
```
Comment (petrspelos) [inline src/Client/Dialogs/ReportDialog.razor]:
```suggestion
</MudDialog>

```

File: src/Client/Extensions/RandomExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+        if (enumerable is null)
+        {
+            throw new ArgumentNullException(nameof(enumerable));
+        }
+
+        var list = enumerable as IList<T> ?? enumerable.ToList();
+
+        return list.Count == 0 ? default! : list[random.Next(0, list.Count)];
+    }
+}
```
Comment (petrspelos) [inline src/Client/Extensions/RandomExtensions.cs]:
```suggestion
}

```

File: src/Client/wwwroot/js/saveAsFile.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+﻿
+function saveAsFile(filename, bytesBase64) {
+    var link = document.createElement('a');
+    link.download = filename;
+    link.href = "data:application/octet-stream;base64," + bytesBase64;
+    document.body.appendChild(link); // Needed for Firefox
+    link.click();
+    document.body.removeChild(link);
+}
```
Comment (petrspelos) [inline src/Client/wwwroot/js/saveAsFile.js]:
```suggestion
}

```

**Review by petrspelos — CHANGES_REQUESTED** [review]
Nothing obviously wrong with it. ☺️ 
Just a couple of missing newline. I also agree with @fpindej's expression body suggestion.

Let's make these changes and get this thing in! 💯 🎉

**Comment by fpindej** [comment]
@thezerda If You don't mind, I will commit @petrspelos' suggestions and will get this into production asap 😊 Gotta try the new deployment pipeline!

**Comment by fpindej** [comment]
@petrspelos Re-review please 🕵🏻

**Review by petrspelos — APPROVED** [review]

**Pull request merged**